### DISCUSSION TITLE
Overriding theme color variable started to require `!important`

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#5635] [Q&A] [answered]
Hi,
I have upgraded my mkdocs site from 4.32.4 to 4.36.0 and noticed, that the way I customized the colors for certain elements stopped working.

Here is an example in my custom CSS that changes an `a` element color:

```css
[data-md-color-scheme="slate"] {
  --md-typeset-a-color: #00c9ff;
```

It works fine in 4.32.4, but when I upgraded to 4.36.0 this variable is overridden by the theme default values:
<img width="324" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/5679861/42d5a44a-e61d-43d0-b61d-c8e1c485a22f">

To overcome this, I had to use `!important` in my CSS next to the variable value. I wonder if this is intended.

**Comment by squidfunk** [comment]
Thanks for asking. Yes, we had to make some changes to make colors more stable across schemes, specifically because of https://github.com/squidfunk/mkdocs-material/pull/5206, which might make it necessary to adjust some overrides. Changes were done in https://github.com/squidfunk/mkdocs-material/commit/6fe1d556071416122af362064c78873babcfe9cf. Could you please share a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) of your issue? Without this, it's hard to anticipate what's the problem, because we don't know how you configured your site or what else might be interferring.

And no, `!important` should only be a last or temporary resort.

**Reply by hellt** [reply]
Here is the reproduction
If you switch to slate (dark) theme the `<a>` color matches indigo color palette, although I am overriding this var in custom.css

I managed to only override it with !important, but as you said, this is a nogo. Likely I am not using the "right" way of overriding color vars

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11856494/example.zip)

**Reply by hellt** [reply]
@squidfunk just in case this discussion has missed your inbox. In case it is not, I am sorry, not meant to push, just as a reminder in case you missed the reproduction.

**Reply by squidfunk** [reply]
Still in my inbox. I'm a little tight on time right now for private reasons, but I haven't forgotten. Will definitely get back to you.

**Comment by kamilkrzyskow** [comment, marked as answer]
Hi @hellt , on your screenshot you can see that the 
```css
[data-md-color-scheme="slate"][data-md-color-primary="indigo"]
```
rule from the `palette.css` file isn't overridden by your 
```css
[data-md-color-scheme="slate"]
```
rule from the `nokia.css` file.

The rule from the `palette.css` file is more specific than the rule from the custom file, it covers both the `data-md-color-scheme` and `data-md-color-primary`. Instead of using `!important` increase the "range" of your rule to be as explicit as the rule from mkdocs-material, in other words add the `[data-md-color-primary="indigo"]` part.

EDIT: The "range" is in fact specificity https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity

**Reply by kamilkrzyskow** [reply]
In case you want to override any color, not just indigo:
```css
[data-md-color-scheme="slate"][data-md-color-primary]
```

**Reply by hellt** [reply]
Thanks @kamilkrzyskow 
indeed, this works like a charm. If that is intended, I am happy to run like that.

For others, here is what I had to add:

```css
[data-md-color-scheme="slate"][data-md-color-primary="indigo"] {
  --md-typeset-a-color: #00c9ff;
/* <other rules> */
}
```

**Reply by squidfunk** [reply]
Thanks @kamilkrzyskow for chipping in. Indeed, this is the way to go!
